cylc/flow/workflow_files.py

```diff
@@ -1907,7 +1907,8 @@ def get_source_workflow_name(source: Path) -> str:
     else the basename of the given path."""
     for dir_ in get_source_dirs():
         try:
-            return str(source.relative_to(dir_))
+            workflow_name = str(source.relative_to(Path(dir_).expanduser()))
+            return workflow_name
```

Comment:
Need to expand env vars too, I think
```suggestion
            return str(source.relative_to(expand_path(dir_)))
```